packages/portal-marketmaker/src/index.js

```diff
@@ -0,0 +1,407 @@
6","name":"","type":"uint256"}],"name":"orderbook","outputs":[{"internalType":"address","name":"maker","type":"address"},{"internalType":"int128","name":"amountSats","type":"int128"},{"internalType":"uint128","name":"priceTokPerSat","type":"uint128"},{"internalType":"bytes20","name":"scriptHash","type":"bytes20"},{"internalType":"uint256","name":"stakedTok","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"owner","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"uint256","name":"amountSats","type":"uint256"},{"internalType":"uint256","name":"priceTokPerSat","type":"uint256"},{"internalType":"bytes20","name":"scriptHash","type":"bytes20"}],"name":"postAsk","outputs":[{"internalType":"uint256","name":"orderID","type":"uint256"}],"stateMutability":"payable","type":"function"},{"inputs":[{"internalType":"uint256","name":"amountSats","type":"uint256"},{"internalType":"uint256","name":"priceTokPerSat","type":"uint256"}],"name":"postBid","outputs":[{"internalType":"uint256","name":"orderID","type":"uint256"}],"stateMutability":"payable","type":"function"},{"inputs":[{"internalType":"uint256","name":"escrowID","type":"uint256"},{"internalType":"uint256","name":"bitcoinBlockNum","type":"uint256"},{"components":[{"internalType":"bytes","name":"blockHeader","type":"bytes"},{"internalType":"bytes32","name":"txId","type":"bytes32"},{"internalType":"uint256","name":"txIndex","type":"uint256"},{"internalType":"bytes","name":"txMerkleProof","type":"bytes"},{"internalType":"bytes","name":"rawTx","type":"bytes"}],"internalType":"struct BtcTxProof","name":"bitcoinTransactionProof","type":"tuple"},{"internalType":"uint256","name":"txOutIx","type":"uint256"}],"name":"proveSettlement","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"contract IBtcTxVerifier","name":"_btcVerifier","type":"address"}],"name":"setBtcVerifier","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint256","name":"_minConfirmations","type":"uint256"}],"name":"setMinConfirmations","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"newOwner","type":"address"}],"name":"setOwner","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint256","name":"_stakePercent","type":"uint256"}],"name":"setStakePercent","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint256","name":"escrowID","type":"uint256"}],"name":"slash","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[],"name":"stakePercent","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"token","outputs":[{"internalType":"contract IERC20","name":"","type":"address"}],"stateMutability":"view","type":"function"}]`;
```

Comment:
check out how `portal-web` uses typechain

this lets you avoid pasting in ABIs = easier to update when contracts change

also provides a typed API